server/repositories/cmsApi.js

```diff
@@ -1,37 +1,70 @@
-const { getEstablishmentUiId } = require('../utils');
+/* eslint-disable class-methods-use-this */
+const { DrupalJsonApiParams: Query } = require('drupal-jsonapi-params');
+
+const topicsQuery = new Query()
+  .addFields('taxonomy_term--tags', [
+    'drupal_internal__tid',
+    'name',
+    'description',
+  ])
+  .addFields('taxonomy_term--moj_categories', [
+    'name',
+    'description',
+    'field_legacy_landing_page',
+  ])
+  .addFilter(
+    'vid.meta.drupal_internal__target_id',
+    ['moj_categories', 'tags'],
+    'IN',
+  )
+  .addPageLimit(100)
+  .getQueryString();
 
 class CmsApi {
   constructor(jsonApiClient) {
     this.jsonApiClient = jsonApiClient;
   }
 
-  async primaryMenu(prisonId = 0) {
-    const response = await this.jsonApiClient.get(
-      `/jsonapi/node/landing_page?fields[node--landing_page]=title,field_moj_description,drupal_internal__nid,field_moj_prisons`,
+  async getTopics(establishmentName) {
+    if (!establishmentName) {
+      return [];
+    }
+
+    const { data } = await this.jsonApiClient.get(
+      `/jsonapi/prison/${establishmentName}/taxonomy_term?${topicsQuery}`,
     );
-    return this.transform(response?.data || {}, prisonId);
+
+    return this.transform(data || []);
```

Comment:
again this could be a bit lame - can't foresee why drupal would return us a falsey value... maybe should hard fail here?

Reply:
As long as the request is considered a valid JSON:API url, then Drupal will always return `data` (and will be empty for no results).  And you will also get a 200 response code.

But there are quite a few other response types, if the url isn't valid.

- `/jsonapi/some-unknown-entity-type` will return a 404 (with an `errors` property and no `data`)
- `/jsonapi/taxonomy_term?filter[some-uknown-field]=1` will return a 400 (with an `errors` property and no `data`)
- `/jsonapi/taxonomy_term?filter=incorrect-way-to-use-a-filter` will return a 500 (this appears to be a bug in Drupal, as it's not able to deal with the request at all and results in a fatal error).

Reply:
cool, I've simplified this, I think any non 2xx/3xx response will result in an exception and a user facing error being presented.